net-utils/src/lib.rs

```diff
@@ -147,46 +143,90 @@ pub fn verify_reachable_ports(
         return ok;
     }
 
-    for _udp_retries in 0..5 {
-        // Wait for a datagram to arrive at each UDP port
-        for udp_socket in udp_sockets {
-            let port = udp_socket.local_addr().unwrap().port();
-            let udp_socket = udp_socket.try_clone().expect("Unable to clone udp socket");
-            let (sender, receiver) = channel();
-            std::thread::spawn(move || {
```

Comment:
Well, not `join()`-ing is a bad taste. I'm pretty sure this leaks a thread or it reads an actual data from the socket at arbitrary time after the validator really start to boot...

Reply:
- [ ] Oops, I'll fix this for tcp as well...